pageserver/src/disk_usage_eviction_task.rs

```diff
@@ -807,14 +807,21 @@ async fn collect_eviction_candidates(
             if !tl.is_active() {
                 continue;
             }
-            let info = tl.get_local_layers_for_disk_usage_eviction().await;
+            let Some(info) = tl.get_local_layers_for_disk_usage_eviction().await else {
+                // if we cannot access the layers right now, we'll try again later.
+                continue;
+            };
             debug!(tenant_id=%tl.tenant_shard_id.tenant_id, shard_id=%tl.tenant_shard_id.shard_slug(), timeline_id=%tl.timeline_id, "timeline resident layers count: {}", info.resident_layers.len());
             tenant_candidates.extend(info.resident_layers.into_iter());
             max_layer_size = max_layer_size.max(info.max_layer_size.unwrap_or(0));
 
             if cancel.is_cancelled() {
                 return Ok(EvictionCandidates::Cancelled);
             }
+
+            // because we no longer do waiting on `RwLock::read` and so make tokio::coop progress,
+            // yield after each successful timeline
+            tokio::task::yield_now().await;
```

Comment:
This is pre-emptive per my earlier plans for Layer (#6028) which are now in flux because of asyncification of heavier_once_cell related to walredo work.

Reply:
Well, this is needed regardless on this PR because the current code does not poll any tokio primitive now.